flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -105,6 +106,14 @@
 
 	private final YarnConfiguration yarnConfig;
 
+	private static final String RM_FAIR_SCHEDULER = "org.apache.hadoop.yarn.server.resourcemanager.scheduler.fair.FairScheduler";
+	private static final String RM_SCHEDULER_INCREMENT_ALLOCATION_MB = "yarn.scheduler.increment-allocation-mb";
+	private static final int DEFAULT_RM_SCHEDULER_INCREMENT_ALLOCATION_MB = 1024;
+	private static final String RM_SCHEDULER_INCREMENT_ALLOCATION_VCORES = "yarn.scheduler.increment-allocation-vcores";
+	private static final int DEFAULT_RM_SCHEDULER_INCREMENT_ALLOCATION_VCORES = 1;
+	private static final String RM_RESOURCE_TYPES_INCREMENT_ALLOCATION_MB = "yarn.resource-types.memory-mb.increment-allocation";
+	private static final String RM_RESOURCE_TYPES_INCREMENT_ALLOCATION_VCORES = "yarn.resource-types.vcores.increment-allocation";
```

Comment:
These constants are not related to the main responsibility of `YarnResourceManager`. Let's try to move them to somewhere else and keep `YarnResourceManager` simple and clear.

I would suggest to introduce a public static util method in `Utils` for creating the `WorkerSpecContainerResourceAdapter`, something like the following.
```
public static createWorkerSpecContainerResourceAdapter(Configuration flinkConfig, YarnConfiguration yarnConfig) {
	// ...
}
```